octobot_trading/exchanges/connectors/ccxt_exchange.py

```diff
@@ -271,124 +308,699 @@ async def get_order_book(self, symbol: str, limit: int = 5, **kwargs: dict) -> t
+        except Exception as e:
+            self.logger.exception(e, True, f"Failed to load positions using private_get_position_risk")
+        return []
+
+    async def get_positions_linear(self, symbol=None, **kwargs: dict) -> list:
+        return await self.get_position_by_sub_type(
+            "linear", symbol,
+            await self.exchange_manager.exchange.get_positions_linear_settle_coins()
+            , **kwargs)
+
+    async def get_positions_inverse(self, symbol=None, **kwargs: dict) -> list:
+        return await self.get_position_by_sub_type(
+            "inverse", symbol,
+            await self.exchange_manager.exchange.get_positions_inverse_settle_coins()
+            , **kwargs)
+
+    async def get_positions_swap(self, symbol=None, **kwargs: dict) -> list:
+        return await self.get_position_by_sub_type("swap", symbol, **kwargs)
+
+    async def get_positions_option(self, symbol=None, **kwargs: dict) -> list:
+        return await self.get_position_by_sub_type("option", symbol, **kwargs)
+
+    async def get_position_by_sub_type(
+            self, sub_type, symbol, settle_coins: list = None, **kwargs: dict
+    ) -> list:
+        params = {**kwargs, "subType": sub_type}
+        if settle_coins:
+            positions = []
+            for coin in settle_coins:
+                params["settleCoin"] = coin
+                if symbol:
+                    positions += await self._get_symbol_positions(symbol, **params)
+                else:
+                    positions += await self._get_positions(**params)
+            return positions
+        if symbol:
+            return await self._get_symbol_positions(symbol, **params)
+        return await self._get_positions(**params)
+
+    async def _get_positions(self, **kwargs: dict) -> list:
+        try:
+            raw_positions = await self.client.fetch_positions(params=kwargs)
+            return await self.exchange_manager.exchange.parse_positions(raw_positions)
+        except Exception as e:
+            self.logger.exception(e, True, "Failed to load positions using get_position_default "
+                                           f"{f'with params: {kwargs}' if kwargs else ''}")
+            return []
 
     async def get_symbol_positions(self, symbol: str, **kwargs: dict) -> list:
+        defined_methods = self.connector_config.GET_SYMBOL_POSITION_METHODS
+        positions = []
+        if self.get_positions_linear.__name__ in defined_methods:
+            positions += await self.get_positions_linear(symbol, **kwargs)
+        if self.get_positions_inverse.__name__ in defined_methods:
+            positions += await self.get_positions_inverse(symbol, **kwargs)
+        if self.get_positions_swap.__name__ in defined_methods:
+            positions += await self.get_positions_swap(symbol, **kwargs)
+        if self.get_positions_option.__name__ in defined_methods:
+            positions += await self.get_positions_option(symbol, **kwargs)
```

Comment:
Same here, we should ask for a specific position type, not just returning what we found on the exchange in my opinion

Reply:
its changed, now its up to the tentacle